src/app/qgsdiscoverrelationsdlg.cpp

```diff
@@ -43,6 +43,15 @@ void QgsDiscoverRelationsDlg::addRelation( const QgsRelation &rel )
   mRelationsTable->setItem( row, 2, new QTableWidgetItem( rel.fieldPairs().at( 0 ).referencingField() ) );
   mRelationsTable->setItem( row, 3, new QTableWidgetItem( rel.referencedLayer()->name() ) );
   mRelationsTable->setItem( row, 4, new QTableWidgetItem( rel.fieldPairs().at( 0 ).referencedField() ) );
+  if ( rel.strength() == QgsRelation::RelationStrength::Composition )
+  {
+    mRelationsTable->setItem( row, 5, new QTableWidgetItem( QStringLiteral( "Composition" ) ) );
```

Comment:
Could you add a tooltip here that explains the effect of setting this to "Composition" or "Association"

Reply:
Ok. And what about the QgsRelationAddDlg? Because there you configure it. What would be good text. Something like this: "Association(loose relation) or Composition (fix relation - child features will be copied to)"

Reply:
Right, where it's configured it's even more important.

And concerning the text: the effect (copying of children) sounds like the most important information.